lib/internal/url.js

```diff
@@ -320,6 +320,64 @@ class URL {
           onParseError);
   }
 
+  static from(input) {
```

Comment:
Can you do `static from(input = {})` instead of the below line? it will prevent stuff like `from(false)` and `from(null)` and `from('')` (and then possibly add tests that all those throw?)

Reply:
Yeah, I was thinking that forming a URL from `false` which is fine right now (if you look up the tests) is not a good idea. I think I will do exactly as you suggest and throw a TypeError.